### ISSUE TITLE
tune is not supported for vault_gcp_auth_backend resource

### ISSUE THREAD

**Issue by ghost** [hashicorp/terraform-provider-vault#882]
Hi there,

When I'm enabling GCP auth method. I want to tune token_ttl however it do not support tune parameter 

Code 
```
resource "vault_gcp_auth_backend" "gcp_auth_backend" {
  credentials  = "JUST TEST"
  path = "gcp"

  tune {
    max_lease_ttl      = "1h"
    default_lease_ttl  = "1h"
  }
}
```
Error
```
$ terraform apply 

Error: Unsupported block type

  on main.tf line 5, in resource "vault_gcp_auth_backend" "gcp_auth_backend":
   5:   tune {

Blocks of type "tune" are not expected here.
```

I was expecting to see something seemlier  as resource `vault_jwt_auth_backend` which supports tune argument

**Comment by papovyr** [comment]
What will be best option here to tune auth backend ?

**Comment by amhn** [comment]
You could import this backend as a generic backend and use tune there.

```
resource "vault_auth_backend" "gcp_auth_backend" {
  path = "gcp"

  tune {
    max_lease_ttl      = "1h"
    default_lease_ttl  = "1h"
  }
}
```
After `terraform import vault_auth_backend.gcp_auth_backend gcp` you should be able to configure it.

This is only a workaround because with both vault_auth_backend and vault_gcp_auth_backend an initial apply without import will fail.

**Comment by papovyr** [comment]
@amhn thanks for response
If i use 
```
resource "vault_auth_backend" "gcp_auth_backend" {
  path = "gcp"

  tune {
    max_lease_ttl      = "1h"
    default_lease_ttl  = "1h"
  }
}
```
How do i configure GCP credentials ? 

If i use generic endpoint I need to match format of JSON for GCP service account JSON which terraform doesn't support 

Will be best to support tune on `vault_gcp_auth_backend` resource

**Comment by amhn** [comment]
Sorry, my answer was not complete. You would use both backend types for the same path, apply one and import it as the other.
```
resource "vault_gcp_auth_backend" "gcp_auth_backend" {
  credentials  = "JUST TEST"
  path = "gcp"
}

resource "vault_auth_backend" "gcp_auth_backend" {
  path = "gcp"

  tune {
    max_lease_ttl      = "1h"
    default_lease_ttl  = "1h"
  }
}
```
Then `terraform apply -target vault_gcp_auth_backend.gcp_auth_backend` and `terraform import vault_auth_backend.gcp_auth_backend gcp`

The problem is, this configuration can not be applied in one step. But after both configs are in the state, I did not notice any conflicts (with the ldap backend). I created this config by importing a configured vault and only noticed the problem after trying to configure a new vault from scratch.
